### DISCUSSION TITLE
绳网帖子大量删除事件最新调查结果与紧急通知

### DISCUSSION THREAD

**Discussion by share121** [share121/inter-knot#4569] [公告]
![1bad5c9e16c86383df334774532c53e](https://github.com/user-attachments/assets/6e20f2e9-ba20-4232-8518-ee6b5d6112d2)

![1bad5c9e16c86383df334774532c53e](https://github.com/user-attachments/assets/6e20f2e9-ba20-4232-8518-ee6b5d6112d2)

![image](https://github.com/user-attachments/assets/3e0ac949-fef5-4024-aca8-9ee1edbe64fd)

经过调查，有用户恶意使用拥有管理员权限的 api 删除帖子

## 原因

公共 api 实际上是 root api，拥有管理员权限。
由于代码开源，我的 Github Apps 中的 rsa 私钥、client\_id、client\_secret 都是公开的。攻击者可以利用这些数据，伪造出删除请求，删除 discussion。

## 解决方法

我已关闭 Github Apps 中的 write 权限，现在无法利用 root api 删除 discussion。

## 通知

旧版即将废弃，请尽快升级到最新版。
你可能需要手动升级。

**Comment by Dave-12138** [comment]
怎么会有人把私钥传到公开仓库的<img src="https://bbs-static.miyoushe.com/static/2024/02/02/2d1279488bab83fea29406150c89f136_1145549606755116007.png" alt="寻梦者-等等" width="75" >

**Reply by share121** [reply]
我的软件需要使用它来生成访问令牌，所以不得不上传

**Reply by hk0852** [reply]
该怎么说呢？他的目的和这种手段不在于，对于这个网站内容的破坏而是在破坏生态，破坏用户对于你的网站的安全以及对内容保护程度的信任度，从而实现类似于绝灭大君幻胧从内部渗透破坏人心安定的作用

**Reply by wyc-26** [reply]
> 我的软件需要使用它来生成访问令牌，所以不得不上传

GitHub Secrets：？？？当我不存在是吧（

**Comment by hk0852** [comment]
该怎么说呢？他的目的和这种手段不在于，对于这个网站内容的破坏而是在破坏生态，破坏用户对于你的网站的安全以及对内容保护程度的信任度，从而实现类似于绝灭大君幻胧从内部渗透破坏人心安定的作用

**Comment by hk0852** [comment]
（对方会不会有更激进的动作？）

**Reply by share121** [reply]
我现在只保留了读取权限，所以他再怎么搞都没法删除帖子了

**Reply by xddxdd** [reply]
现在这样也不保险，因为私钥总要嵌入到编译出的程序中的，可以逆向出来。建议给用户的私钥和token就不要带过多权限了，要做管理操作时专门用一个只有自己知道的私钥/token编译一份自用，或者直接通过GitHub平台来操作。

**Reply by xddxdd** [reply]
当然最好的办法还是用cloudflare workers之类的serverless function写一个服务，在不暴露私钥的情况下给客户端提供access token。

**Comment by momomomofish** [comment]
看不懂细节，总之大佬牛逼👍

**Comment by obeyears** [comment]
怪不得一看我水水的disscussion没了🤔

**Comment by EhuanyingK** [comment]
真的缺德，我的两个30+评论的帖子都没了

**Comment by Wodlie** [comment]


**Reply by momomomofish** [reply]
这种生物能学会用电脑，说明我国生态恢复的挺好

**Reply by WinterSamu** [reply]
话说这些不明生物怎么跟曱甴一样恶心人，到哪哪都有<img src="https://upload-bbs.miyoushe.com/upload/2023/01/18/06f3f4c54fc51d82ba3e575194504d80_6167770388617778929.png" alt="米游君-喝可乐" width="75" >

**Comment by hk0852** [comment]
还是人工手磨味道更纯，他自己本人应该不会弄，但是他应该是像当年原开服的时候那个快斯的东西请了开合勾来爆破，这种生物现实生活里都不是能正常存活的东西，开合勾或者那种脚本小子他们会干这种事只是因为干了这种事之后满足了他们内心的变态快乐感，雇主给钱，并且快乐，所以就直接大胆干了

**Comment by 4B47-DA27** [comment]
好像可以说话了

**Comment by Junjun280** [comment]
好多宝藏帖子都没了（悲)<img src="https://img-static.mihoyo.com/communityweb/upload/12a80a8f5da3949b086c497b8cc4b5d3.png" alt="安比_头痛" width="75" >

**Comment by XinRenShenJiangZiShu** [comment]
所以我这个刚弄完杂七杂八东西的新人绳匠，又要更新！

**Comment by KAFKA147** [comment]
怎么说呢，肯定是米黑子干的

**Comment by Tragedysaver1024** [comment]
我本来看到这种人之前还是挺相信达尔文的进化论的，直到我发现自然选择这只大手不能很好的过滤一些伟大种群的废料

**Comment by HLXS-HLXS** [comment]
事实证明，绳网门槛还是太低了<img src="https://bbs-static.miyoushe.com/static/2023/12/22/4da6b5aa297be428e0c4c478c33d11db_1621822785592350465.png" alt="月下-威胁" width="75" >

**Comment by Yizakl** [comment]
666有简笔
